hedging/src/job/adjust_funding.rs

```diff
@@ -0,0 +1,222 @@
+
+use crate::{error::*, okex_transfers::*, rebalance_action::*, synth_usd_liability::*};
+
+use super::HedgingFundingConfig;
+
+const SATS_PER_BTC: Decimal = dec!(100_000_000);
+
+#[instrument(name = "adjust_funding", skip_all, fields(correlation_id = %correlation_id,
+        target_liability, current_position, last_price_in_usd_cents, funding_available_balance,
+        trading_available_balance, onchain_fees, action, external_client_transfer_id, internal_client_transfer_id,
+        transferred_funding) err)]
+pub(super) async fn execute(
+    correlation_id: CorrelationId,
+    synth_usd_liability: SynthUsdLiability,
+    okex: OkexClient,
+    okex_transfers: OkexTransfers,
+    galoy: GaloyClient,
+    hedging_funding_config: HedgingFundingConfig,
+) -> Result<(), HedgingError> {
+    let span = tracing::Span::current();
+
+    let target_liability_in_cents = synth_usd_liability.get_latest_liability().await?;
+    span.record(
+        "target_liability",
+        &tracing::field::display(target_liability_in_cents),
+    );
+
+    let current_position = okex.get_position_in_signed_usd_cents().await?;
+    span.record(
+        "current_position",
+        &tracing::field::display(current_position.usd_cents),
+    );
+
+    let mut last_price_in_usd_cents = current_position.last_price_in_usd_cents;
+    if last_price_in_usd_cents.is_zero() {
+        last_price_in_usd_cents = okex.get_last_price_in_usd_cents().await?.usd_cents;
+    }
+
+    span.record(
+        "last_price_in_usd_cents",
+        &tracing::field::display(last_price_in_usd_cents),
+    );
+
+    let funding_available_balance = okex.funding_account_balance().await?;
+    span.record(
+        "funding_available_balance",
+        &tracing::field::display(&funding_available_balance),
+    );
+
+    let trading_available_balance = okex.trading_account_balance().await?;
+    span.record(
+        "trading_available_balance",
+        &tracing::field::display(&trading_available_balance),
+    );
+
+    let fees = okex.get_onchain_fees().await?;
+    span.record("onchain_fees", &tracing::field::display(&fees));
+
+    let is_internal_transfer_pending = okex_transfers.is_internal_transfer_pending().await?;
+    span.record(
+        "is_internal_transfer_pending",
+        &tracing::field::display(&is_internal_transfer_pending),
+    );
+
+    let is_external_transfer_pending = okex_transfers.is_external_transfer_pending().await?;
+    span.record(
+        "is_external_transfer_pending",
+        &tracing::field::display(&is_external_transfer_pending),
+    );
+
+    let fees = okex.get_onchain_fees().await?;
+    span.record("onchain_fees", &tracing::field::display(&fees));
+
+    let action = determine_action(
+        target_liability_in_cents,
+        current_position.usd_cents.into(),
+        trading_available_balance.total_amt_in_btc,
+        last_price_in_usd_cents,
+        funding_available_balance.total_amt_in_btc,
+        is_internal_transfer_pending,
```

Comment:
I don't think `determine_action` should depend on this. It should get the current state and determine what to do. Anything inflight is irrelevant. If there is something in flight the database won't reserve a 'slot' and no action will be performed. Hence we already have an implicit `DoNothing` whenever something is in flight.

The only thing that would make sense here is to input the `amount` that is inflight. Then we could figure out if additional action should be taken. But that makes the logic more complicated. The idea of the `slot` reservation is to simplify things and guarantee that only 1 thing is happening at a time and nothing will get kicked off if something is already in flight.

The logic in `determin_action` should become simpler without taking in flight into account.